### ISSUE TITLE
Is it possible to not add dependence styled-components?

### ISSUE THREAD

**Issue by fourteenmeister** [based-ghost/react-functional-select#3]

**Comment by based-ghost** [comment]
No, not at the moment at least. However, looking at the roadmap for v5 of styled-components made it the best choice of all of the CSS-in-JS libraries in my opinion (in terms of performance and bundle-size). Version 5 is currently in a release-candidate state but it is stable and I have been using it in various projects with no issues, so, if you are worried about how it contributes to your bundle size v5 has a 31% smaller bundle-size than v4:

`npm i styled-components@5.0.0-rc.2`

**Comment by StarpTech** [comment]
Hi,
I just wanted to open the same issue. The lib looks really promising. I don't want to be forced to use a specific CSS-in-JS solution. I only want to reuse the **logic** of a select component. @based-ghost what do you think about this approach https://github.com/JedWatson/react-select/issues/2706#issuecomment-614794918

**Comment by based-ghost** [comment]
@StarpTech - Ahhh, yes in retrospect I wish I would have taken the time to build this out in a more modular way upfront, so that consumers of the package could opt-out of styled-components (thereby, avoiding the need to eat that dependency). At the time I started deving this lib the `React Hooks API` was fairly new, so writing the lib without the use of classic class components + providing the opt-out of CSS-in-JS functionality proved to be a little more than I was able to handle at the time.

Your comment reminded me to take another look at this though, since this is something I did want to investigate and there is probably a way to smoothly integrate this with the current lib. I wrote this lib in a way that would make it flexible for additions/changes like this - so, hopefully that proves to be the case with this functionality.

I'll keep you posted and also take a look into how `react-select` handled this - as that package was a source of inspiration for many other features I incorporated.

**Comment by StarpTech** [comment]
Hi @based-ghost thanks for the feedback and keep up the good work.
